CTC: update spouse prior tax year questions

wschaefer opened the pull request:
Co-authored-by: Travis Grathwell <tgrathwell@codeforamerica.org>

bytheway875 commented:
@wschaefer can you take a look at the method `spouse_prior_tax_year_agi` as used on the return_header1040.rb and confirm that that logic is still correct based on the updates in this PR?

jenny-heath commented:
possible changes to make to the portal (later?):
- display spouse prior year AGI if filing jointly _this year_
- allow client to edit spouse prior year AGI if filed separately and spouse filed full return _last year_ (in other words, only allow to edit when we are not inferring the amount ourselves)

wschaefer commented:
@bytheway875 Thanks! We also liked the idea of removing the duplication but ended up going the other direction (not saving extra values for the amount) instead of removing the method that calculates the amount.

Our thought was that it would be harder to keep the data accurate if we updated spouse_prior_year_agi_amount with calculated amounts (ie if we set it to be primary_prior_year_agi_amount and then the primary amount changes we'd have to remember to update the spouse amount). If we just keep spouse_prior_year_agi_amount as the amount the client enters (if they do) and keep spouse_filed_prior_tax_year then its easy to always get the right amount. We moved that logic into a method on the intake so we can use it more places.
